### ISSUE TITLE
Experiencing unhealthy snowman consensus with geographically distributed nodes around clusters of addValidator and removeValidator events

### ISSUE THREAD

**Issue by l-monninger** [ava-labs/avalanchego#2713]
**Describe the bug**
We've been running a permissioned fuji subnet with between 5 and 9 validators for a couple days. We've run into `unhealthy snowman consensus` several times after submitting several `addValidator` and `removeValidator` transactions via the Avalanche CLI within the same hour. Usually, this is accompanied by varying errors on different peers:
- `router error`
- `grpc client connection error`

**To Reproduce**
I've attached archives of logs for three of the nodes, titled by their regions.

**Expected behavior**
I may be missing some parameter that needs to be tuned. But, I wasn't expecting my subnet to be some brittle to these changes in the validator set. 

**Screenshots**
N/A

**Logs**
[sa-east-1.tar.gz](https://github.com/ava-labs/avalanchego/files/14202450/sa-east-1.tar.gz)
[eu-central-1.tar.gz](https://github.com/ava-labs/avalanchego/files/14202451/eu-central-1.tar.gz)
[us-west-1.tar.gz](https://github.com/ava-labs/avalanchego/files/14202452/us-west-1.tar.gz)

**Metrics**
These can be provided if the logs are not sufficient. 

**Operating System**
All nodes are running on Ubuntu 22.04

**Additional context**
- The first time we saw something like this, we thought it had something to do with adding a validator before it had bootstrapped with a validator set of six (then seven) nodes. 

**To best protect the Avalanche community security bugs should be reported in accordance to our [Security Policy](../security/policy)**

**Comment by joshua-kim** [comment]
Seeing a panic in the logs, can you confirm that this is coming from your VM code? If your VM is panick'ing that would explain why the connection is being closed unexpectedly:

```
thread 'tokio-runtime-worker' panicked at subnet/src/vm/mod.rs:986:68:
called `Result::unwrap()` on an `Err` value: InternalError { error: "Gap in epoch history. Trying to put in LedgerInfo in epoch: 1, current epoch: 4" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by l-monninger** [comment]
Yeah, that's coming from us.

**Comment by joshua-kim** [comment]
I think you will have to debug the reason your VM is panicking. From avalanchego's perspective i think this is expected behavior as losing connection with the VM process is considered a fatal event.

**Comment by l-monninger** [comment]
Yeah, I agree. Even without any panicks this remains problem though. We may need a different reorg strategy.

**Comment by aaronbuchwald** [comment]
For the reorg strategy, I'm not familiar with your code, so can't attest to exactly what the issue is, but the consensus engine maintains these invariants for the VM in terms of the order that blocks will be inserted/accepted/rejected: https://github.com/ava-labs/avalanchego/tree/master/vms#snowman-vms.

**Comment by l-monninger** [comment]
For completeness, the associated issue on our end is here: https://github.com/movemntdev/M1/issues/104

**Comment by aaronbuchwald** [comment]
Thanks for linking!
